neu-self-image-experiments/sie-backend
# SIE-70 Refactored files into python packages and added storage trigger to cloud run

Apologies for a lengthy PR...

Here is what I did:
- Deleted `archive` directory -> This is to make our container as lightweight as possible
- Moved some of @xuzhouyuan README stuff -> again to make things light
- Refactored files into python packages for maintainability/extensibility
- Removed comments & redundant code
- Added comments (more will be added)
- Set up a gunicorn server that ONLY listens internally (i.e. not public) for storage trigger from `sie-raw-images` bucket

How to review this PR?
- Please look at `stimuli_ci_generation` directory

How to test?
- I'm making this PR to first test my trigger with cloud run. If implemented correctly, the container will be continuously deployed and listen to event triggers from `sie-raw-images`
- Update: it works!! Try uploading a file to `sie-raw-images` and head over to cloud run logs to check it out!

Next step:
- Run image processing upon receiving the event trigger
